kie-internal/src/main/java/org/kie/internal/task/api/TaskPersistenceContext.java

```diff
@@ -32,14 +33,23 @@
 import org.kie.internal.task.api.model.ContentData;
 import org.kie.internal.task.api.model.Deadline;
 import org.kie.internal.task.api.model.FaultData;
+import org.kie.internal.task.api.model.Operation;
 
 public interface TaskPersistenceContext {
 
     Task findTask(Long taskId);
 
     Task persistTask(Task task);
-
-    Task updateTask(Task task);
+    
+    default Task updateTask (Task task) {
+        return updateTask (task, Operation.Update);
+    }
+    
+    default Task updateTask (Task task, Operation operation) {
+        return updateTask (task, operation, Optional.empty());
+    }
+
+    Task updateTask (Task task, Operation operation, Optional<String> targetEntity);
```

Comment:
As an internal API, doesn't need to be defined as default, does it?

Reply:
Im not using default here to achieve backward compatibility, but to provide default implementations that I feel will be valid by any implementation